File: gc-commons/gc-base-common/src/main/java/com/gc/common/base/imports/EnableSpringContext.java
package com.gc.common.base.imports;

import com.gc.common.base.spring.ApplicationContextRegister;
import org.springframework.context.annotation.Import;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

/**
 * spring容器引入类
 * @author jackson
 */
@Target(ElementType.TYPE)
@Retention(RetentionPolicy.RUNTIME)
@Import(ApplicationContextRegister.class)
public @interface EnableSpringContext {
}

File: smart-commons/smart-common/src/main/kotlin/com/smart/common/import/EnableSpringContext.kt
package com.smart.common.import

import com.smart.common.spring.ApplicationContextRegister
import org.springframework.context.annotation.Import

/**
 * 引入spring容器
 * @author ming
 * 2019/7/6 下午8:31
 */
@Import(ApplicationContextRegister :: class)
annotation class EnableSpringContext {
}